product-pricing-service/src/main/java/com/hedvig/contract/shared/aggregates/BaseContract.kt

```diff
@@ -395,14 +395,24 @@ abstract class BaseContract<TAgreement : BaseAgreement> : ContractStereotype<TAg
   }
 
   @CommandHandler
-  fun handle(command: TerminateContract) {
-    if (command.terminationDate == terminationDate) return
-    ensureContractIsActive(command)
-    ensureTerminationDateNotInPastMonths(
-      terminationDate = command.terminationDate,
-      dangerouslyAllowHistoryChange = command.dangerouslyAllowHistoryChange,
-      command = command
-    )
+  fun handle(command: TerminateContract): TerminateContractResponse {
+    if (command.terminationDate == terminationDate) return TerminateContractResponse.ALREADY_TERMINATED
+    try { ensureContractIsActive(command) }
+    catch (exception: Exception) {
```

Comment:
The exception type is known here. They all throw specific excpetions
```suggestion
    catch (exception: IllegalWhenContractIsAlreadyTerminatedException) {
```